### ISSUE TITLE
Query on integrating neuromorphic datasets

### ISSUE THREAD

**Issue by ozilbbg11** [fangwei123456/spikingjelly#336]
Hi,
I'm new to SpikingJelly and currently following the tutorial on [Classify DVS Gesture](https://spikingjelly.readthedocs.io/zh_CN/latest/activation_based_en/classify_dvsg.html), and I have self-collected event data (also some gesture recognition) that is nearly identical to the tutorial's data, except for differences in resolution ( `640*480` for mine). I'm wondering if there is a more generic method to integrate new neuromorphic datasets that are slightly different, rather than having to modify multiple python files. Or do you think the network structure on this tutorial won't suit other datasets, and I have to build a new neural network from sketch anyway?

**Comment by fangwei123456** [comment]
>I'm wondering if there is a more generic method to integrate new neuromorphic datasets that are slightly different, rather than having to modify multiple python files

Yes. As you say, the only difference is resolution. We can modify this function to change height and width:

https://github.com/fangwei123456/spikingjelly/blob/410bf7edb14901d7614de80e863e706d57e70edd/spikingjelly/datasets/dvs128_gesture.py#L308

However, the data pre-processing of DVS Gesture is very different with other datasets. The main difference is that in other datasets, one file is one sample. While in DVS Gesture, one file is many samples. Refer to:

https://github.com/fangwei123456/spikingjelly/blob/410bf7edb14901d7614de80e863e706d57e70edd/spikingjelly/datasets/dvs128_gesture.py#L35

If your custom dataset use `one file one sample`, I suggest that you can read codes of https://github.com/fangwei123456/spikingjelly/blob/410bf7edb14901d7614de80e863e706d57e70edd/spikingjelly/datasets/cifar10_dvs.py#L109 .

By the way, if your dataset will be public, I can help to add it to SpikingJelly.

**Comment by ozilbbg11** [comment]
Thanks for your reply. And if I change the height and width
https://github.com/fangwei123456/spikingjelly/blob/410bf7edb14901d7614de80e863e706d57e70edd/spikingjelly/datasets/dvs128_gesture.py#L308
Do I need to modify the structure of the model in this class?
https://github.com/fangwei123456/spikingjelly/blob/410bf7edb14901d7614de80e863e706d57e70edd/spikingjelly/activation_based/model/parametric_lif_net.py#L122

> By the way, if your dataset will be public, I can help to add it to SpikingJelly.

I don't think so, they are just some tiny datasets for my schoolwork.

**Comment by fangwei123456** [comment]
> Do I need to modify the structure of the model in this class?

Yes, the origin model is designed for 2x128x128 input. If you change input size, you will get an error at the first linear layer for size dismatching.
